gssapi/tests/test_raw.py

```diff
@@ -629,25 +671,41 @@ def test_rfc5587(self):
             for expected_mech in expected_mechs:
                 self.assertNotIn(expected_mech, mechs)
 
-        for attr, expected_mechs in known_attrs_dict.items():
-            attrs = set([attr])
+        if self.realm.provider.lower() != 'heimdal':
+            # Heimdal doesn't fully implement gss_indicate_mechs_by_attrs
+            for attr, expected_mechs in known_attrs_dict.items():
+                attrs = set([attr])
 
-            mechs = gb.indicate_mechs_by_attrs(None, None, attrs)
-            self.assertGreater(len(mechs), 0)
-            self.assertEqual(mechs, expected_mechs)
+                mechs = gb.indicate_mechs_by_attrs(None, None, attrs)
+                self.assertGreater(len(mechs), 0)
+                self.assertEqual(mechs, expected_mechs)
 
     @ktu.gssapi_extension_test('rfc5587', 'RFC 5587')
     def test_display_mech_attr(self):
-        test_attrs = [
-            # oid, name, short_desc, long_desc
-            # Taken from krb5/src/tests/gssapi/t_saslname
-            [gb.OID.from_int_seq("1.3.6.1.5.5.13.24"), b"GSS_C_MA_CBINDINGS",
-             b"channel-bindings", b"Mechanism supports channel bindings."],
-            [gb.OID.from_int_seq("1.3.6.1.5.5.13.1"),
-             b"GSS_C_MA_MECH_CONCRETE", b"concrete-mech",
-             b"Mechanism is neither a pseudo-mechanism nor a composite "
-             b"mechanism."]
-        ]
+        test_attrs = []
+        if self.realm.provider.lower() == 'mit':
+            test_attrs = [
+                # oid, name, short_desc, long_desc
+                # Taken from krb5/src/tests/gssapi/t_saslname
+                [gb.OID.from_int_seq("1.3.6.1.5.5.13.24"),
+                 b"GSS_C_MA_CBINDINGS", b"channel-bindings",
+                 b"Mechanism supports channel bindings."],
+                [gb.OID.from_int_seq("1.3.6.1.5.5.13.1"),
+                 b"GSS_C_MA_MECH_CONCRETE", b"concrete-mech",
+                 b"Mechanism is neither a pseudo-mechanism nor a composite "
+                 b"mechanism."]
+            ]
+        elif self.realm.provider.lower() == 'heimdal':
+            test_attrs = [
+                # Taken from lib/gssapi/mech/gss_oid.c
+                [gb.OID.from_int_seq("1.3.6.1.5.5.13.24"),
+                 b"GSS_C_MA_CBINDINGS", b"channel-bindings",
+                 b""],
+                [gb.OID.from_int_seq("1.3.6.1.5.5.13.1"),
+                 b"GSS_C_MA_MECH_CONCRETE", b"concrete-mech",
+                 b"Indicates that a mech is neither a pseudo-mechanism nor a "
+                 b"composite mechanism"]
+            ]
```

Comment:
So these two blocks almost match.  Maybe instead of having two stanzas, you could add a `if heimdal: fix descriptions` condition?